### PR TITLE
Chore: Fix unreliable test in `SpanGraph/index.test.tsx`

### PR DESCRIPTION
**Pull request by CrypticSignal**
[A test ](https://github.com/grafana/grafana/pull/60200/files#diff-fe2c1c8f49f067e863a822fc5d9e9d1ad4de5985ff184b950977d232e30bebaeR64)added in [this PR](https://github.com/grafana/grafana/pull/60200) is currently unreliable:
```
  it('renders <TickLabels /> with the correct value next to each tick', () => {
    render(<TickLabels numTicks={5} duration={1} />);
    expect(screen.getByText(/0.2/)).toBeTruthy();
    expect(screen.getByText(/0.4/)).toBeTruthy();
    expect(screen.getByText(/0.6/)).toBeTruthy();
    expect(screen.getByText(/0.8/)).toBeTruthy();
  });
```
This is because there are sometimes cases where there is more than one element matching the regex and the RTL `getByText` query throws an error in these cases. 

This PR changes the query to `getAllByText` so that the test passes even if there is more than one match.

### CODE REVIEW

**Review by ashharrison90 — COMMENTED** [review]
i would try and fix this in a different way tbh. find something that guarantees you're finding the correct element. for example:
```typescript
import { within } from '@testing-library/react';

...

  it('renders <TickLabels /> with the correct value next to each tick', () => {
    const tickLabels = screen.getByTestId('TickLabels');
    expect(within(tickLabels).getByText(/0.2/)).toBeInTheDocument();
    expect(within(tickLabels).getByText(/0.4/)).toBeInTheDocument();
    expect(within(tickLabels).getByText(/0.6/)).toBeInTheDocument();
    expect(within(tickLabels).getByText(/0.8/)).toBeInTheDocument();
  });
```

also i don't think we should ever be rendering the `TickLabels` component itself in a test about the overall `SpanGraph` component. as you render the main component already in the `beforeEach`, what ends up happening is you're rendering 2 separate components (you can see this if you do `screen.debug()`). if you want to test `TickLabels` in isolation, add a test to the `TickLabels.test.tsx` file.

**Comment by CrypticSignal** [comment]
@ashharrison90 
>i don't think we should ever be rendering the TickLabels component itself in a test about the overall SpanGraph component.

Before I converted this file to RTL we were also rendering the `TickLabels` component in this file as you can see here: https://github.com/grafana/grafana/pull/60200/files#diff-fe2c1c8f49f067e863a822fc5d9e9d1ad4de5985ff184b950977d232e30bebaeL52

So I suppose this is why I decided to test that the `TickLabels` component is rendered with the correct values in this file without giving it much thought.

But I agree with you that tests related to how the `TickLabels` component is rendered should be in `TickLabels.test.tsx` instead. So I'll add the code block you provided to `TickLabels.test.tsx` (and change the expected values) and remove this test from `index.test.tsx`.

**Comment by CrypticSignal** [comment]
@ashharrison90 
There is another test related to rendering `TickLabels` in this file:
```
  it('renders <TickLabels />', () => {
    const tickLabelsComponent = screen.getByTestId('TickLabels');
    expect(tickLabelsComponent).toBeTruthy();
  });
```

But I think we should keep the above test in this file because in `index.tsx`, we render the `TickLabels` component, so it makes sense to me that we test whether it is simply rendered in `index.test.tsx`. Do you agree?

**Comment by ashharrison90** [comment]
yeah i think that all makes sense. as long as we're not rendering a different component, it's fine 👍

**Comment by CrypticSignal** [comment]
> yeah i think that all makes sense. as long as we're not rendering a different component, it's fine 👍

Done!

File: packages/jaeger-ui-components/src/TracePageHeader/SpanGraph/TickLabels.test.tsx [hunk 1]
```diff
@@ -50,4 +50,11 @@ describe('<TickLabels>', () => {
       expect(tick).toHaveStyle(`left: ${pos};`);
     });
   });
+  it('shows the correct value above each tick', () => {
+    expect(screen.getAllByText(/0/)).toBeTruthy();
```
Comment (ashharrison90) [inline packages/jaeger-ui-components/src/TracePageHeader/SpanGraph/TickLabels.test.tsx:54]: again, think we should avoid using `getAllByText`. use a selector that finds the exact element needed.

**Review by ashharrison90 — COMMENTED** [review]
added a comment, otherwise lgtm

File: packages/jaeger-ui-components/src/TracePageHeader/SpanGraph/TickLabels.test.tsx (hunk 1 above)
Comment (CrypticSignal) [reply packages/jaeger-ui-components/src/TracePageHeader/SpanGraph/TickLabels.test.tsx]: I've made this test more strict now. See commit https://github.com/grafana/grafana/pull/61009/commits/8d0e8a83d7e7c7f3ccd78330bb7b11265e48f554.

**Review by ashharrison90 — APPROVED** [review]
lgtm 🙌

**Pull request merged**